### PR TITLE
Better handle plural forms for localization 

### PR DESCRIPTION
**Pull request by DelazJ**
I suggest we switch translation style following https://doc.qt.io/qt-5/i18n-source-translation.html#handling-plurals.

### CODE REVIEW

File: src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp [hunk 1]
```diff
@@ -4093,7 +4093,7 @@ QgsProcessingFieldPanelWidget::QgsProcessingFieldPanelWidget( QWidget *parent, c
 
   if ( mParam )
   {
-    mLineEdit->setText( tr( "%1 fields selected" ).arg( 0 ) );
+    mLineEdit->setText( tr( "%1 field(s) selected", nullptr, 0 ) );
```
Comment (m-kuhn) [inline src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp:4096]:
```suggestion
    mLineEdit->setText( tr( "%n field(s) selected", nullptr, 0 ) );
```

**Comment by m-kuhn** [comment]
Did you test it locally also with English?

**Comment by nyalldawson** [comment]
@m-kuhn 

>  Did you test it locally also with English?

I don't think the pluralisation support works when you run the "default language" build. (at least, it doesn't for me)

**Comment by m-kuhn** [comment]
> @m-kuhn
> 
> > Did you test it locally also with English?
> 
> I don't think the pluralisation support works when you run the "default language" build. (at least, it doesn't for me)

That's my experience too and the reason for this question. I think we need to introduce some English->English translation here.

**Comment by nyalldawson** [comment]
@m-kuhn 
that'd be nice!

**Comment by nirvn** [comment]
@m-kuhn , you are correct, and that'd be nice.

**Comment by DelazJ** [comment]
@m-kuhn TBH I didn't test anything. There are already few places with this syntax so I just applied it to these strings I was about to translate.

**Comment by m-kuhn** [comment]
@DelazJ 
image

Confirmed

**Comment by m-kuhn** [comment]
I find plenty of interesting dialects but no plain `en` on transifex
image

**Comment by DelazJ** [comment]
The QGIS project is currently in `en` so IMHO it makes sense to not find it as target language
image
But I find more English based languages in transifex, namely an en-US, which is the English based locale we expose in QGIS Settings dialog. Maybe there is some switch possible, code-wise...
image

**Comment by m-kuhn** [comment]
Everyone ok if we add a new en_US translation for this purpose (CC @rduivenvoorde )

**Comment by rduivenvoorde** [comment]
I'm probably fine with this, but not fully following...

I'm pretty sure (while translating to dutch 10 years ago...), I've seen those plural forms more around?

So 'en' stays the 'source' lang, but we add an 'en_US' for... translating plurals (so: somebody will have to find all those plural words)?
Or the other way around?
Or something else?

**Comment by m-kuhn** [comment]
What we currently have is the "source code english" which is uploaded to transifex as "en". Saying things like "Deleted 1 feature(s)". This is currently also what ends up on English installations.
What we could have is "Deleted 1 feature", "Deleted 3 features". Which would be a tiny bit nicer. But we need a different language for that, en_US would be a random pick we could use for this purpose.

**Comment by rduivenvoorde** [comment]
Ah, clear. go! It would also help to kill this message :-)

`src/core/qgsapplication.cpp:196 : (QgsApplication) [129ms] loading of qgis translation failed /home/richard/bin/qgis_/master/debug/share/qgis/i18n//qgis_en_US
`

**Comment by m-kuhn** [comment]
Here we go
https://github.com/qgis/QGIS/pull/47204

**Review by m-kuhn — APPROVED** [review]

**Comment by m-kuhn** [comment]
@DelazJ you'll need to have a look at the unit tests which actually test for the changed strings

https://github.com/qgis/QGIS/runs/5062889475?check_suite_focus=true#step:13:156

**Pull request merged**